### PR TITLE
Put dimension symbols into the model signature instead of using -1 and verify inputs using the symbols.

### PR DESCRIPTION
**Pull request by tungld**
Before this PR, dynamic dimensions are printed as -1 in the model signature. This PR uses symbols from dim_params instead of -1. As a result, we can verify that two dimensions with the same symbol must be the same. Code for input verification is also generated.

Example of signature with symbols:
```
Model's input signature:  [    { "type" : "i64" , "dims" : ["dims52" , "dims27"] , "name" : "L_kwargs_input_ids_" }
 ,    { "type" : "i64" , "dims" : ["dims52" , "dims27"] , "name" : "L_kwargs_token_type_ids_" }
 ,    { "type" : "i64" , "dims" : ["dims52" , "dims27"] , "name" : "L_kwargs_attention_mask_" }

]
Model's output signature:  [   { "type" : "f32" , "dims" : ["dims52" , "dims27" , 384] , "name" : "layer_norm_12" }
 ,    { "type" : "f32" , "dims" : ["dims52" , 384] , "name" : "tanh" }
```

### CODE REVIEW

**Review by AlexandreEichenberger — APPROVED** [review]
LGTM, but have 3 observations:
* I would add something in the documentation to explain this feature. It feel that we have an increasing amount of "tests" we do on the inputs, from compiler given LB, UB, and now value checking of same variables. Info on this, plus an example of the error message it gives would certainly help the user.
* as much as I like the `dim35689` values, could we normalize them to dim0, dim1, dim2... Even better, if the onnx input gave names to these dims, could we reuse the dim names? 
* In an example, it would be nice to also have a literal value; I assume (but not sure) that if one dim was `1`, then we would see the literal instead of the `dim234`. And we would check also that the input has indeed `1` if that is the case?. And I would give a different "error" name.

I understand if you don't have the time to do this now before you go, just open a quick issue if you agree with the suggestions and have no time to do it now.

**Comment by tungld** [comment]
@AlexandreEichenberger thank you for your suggestion!

> I would add something in the documentation to explain this feature. It feel that we have an increasing amount of "tests" we do on the inputs, from compiler given LB, UB, and now value checking of same variables. Info on this, plus an example of the error message it gives would certainly help the user.

Yes, added a document about input verification.

> as much as I like the dim35689 values, could we normalize them to dim0, dim1, dim2... Even better, if the onnx input gave names to these dims, could we reuse the dim names?

Sorry for the confusion. `dim35689` here is from the dim_param of the onnx model (generated by pytorch in my example). So symbolic names in the original onnx model are used. We do not change/normalize the dim_params from the original model.

> In an example, it would be nice to also have a literal value; I assume (but not sure) that if one dim was 1, then we would see the literal instead of the dim234. And we would check also that the input has indeed 1 if that is the case?. And I would give a different "error" name.

Yes, static dim has already been handled correctly before this PR, and it was checked. In the backend test for input verification, I added one test for a mix of dynamic and static dimensions.

**Comment by AlexandreEichenberger** [comment]
LGTM; by the way, it would be interesting to attach the names to the dimAnalysis too; for example, we know that typically a BS is small, but a sequence length is growing and large. So this could be used to guide our parallelization heuristics.  The problem is that names are attached to variables; and decisions needs to be taken at the loop level; but using the indices names from the outputs are typically close enough.

**Comment by tungld** [comment]
@jenkins-droid test this please

**Comment by tungld** [comment]
@jenkins-droid test this please
